capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -0,0 +1,255 @@
+import React, { Component } from 'react';
+
+import { Button, Form, Spinner, Modal, Col, Row, Card } from 'react-bootstrap'
+
+import '../styles/SearchUserModal.css'
+import '../styles/Modal.css'
+
+export class SearchUserModal extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      fetchingUsers: false, // For Spinner
+      showModal: false,
+      typingTimeout: 0,
+      searchTerm: '',
+      searchResults: [],
+      addedUsersTracker: [],
+      addedUsers: [],
+      addedFriends: [],
+      resultsFound: false,
+    }
+  }
+
+  getBooks = async (searchTerm) => {
+
+    this.setState({ fetchingUsers: true })
+
+    let searchResults;
+
+    if (searchTerm === '') {
+      searchResults = [];
+
+      this.setState({ searchResults, fetchingUsers: false, resultsFound: false })
+    }
+    else {
+      searchResults = await fetch(`/api/userSearch?searchTerm=${searchTerm}`)
+        .then(response => response.json())
+        .catch(err => alert(err));
+
+      if (typeof searchResults === 'undefined') {
+        searchResults = [];
+      }
+
+      if (searchResults.length !== 0) {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: true }) // Guilty
+      }
+      else {
+        this.setState({ searchResults, fetchingUsers: false, resultsFound: false }) // Not Guilty
```

Comment:
What do "Guilty" and "Not Guilty" mean in this context?

Reply:
We assume that ```resultsFound``` is true. When we finally fetch the searchTerm, if it return an empty array then ```resultsFound``` is guilty and set to false. If ```resultsFound``` is false then we render ```No Results Found```.